frontend/pkg/frontend/cluster.go

```diff
@@ -563,62 +583,67 @@ func (f *Frontend) updateHCPClusterInCosmos(ctx context.Context, writer http.Res
 		return err
 	}
 
+	// TODO is patch supposed to be status accepted?
 	_, err = arm.WriteJSONResponse(writer, http.StatusOK, responseBytes)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func decodeDesiredClusterPatch(ctx context.Context, oldInternalCluster *api.HCPOpenShiftCluster) (*api.HCPOpenShiftCluster, error) {
-	versionedInterface, err := VersionFromContext(ctx)
+// mergeToInternalCluster renders a CS Cluster object in JSON format, applying
+// the necessary conversions for the API version of the request.
+// TODO this overwrite will transformed into a "set" function as we transition fields to ownership in cosmos
+func mergeToInternalCluster(csCluster *arohcpv1alpha1.Cluster, internalCluster *api.HCPOpenShiftCluster) (*api.HCPOpenShiftCluster, error) {
```

Comment:
next three functions are moved here and delegate to each other so they are consistent. There were two slightly different flows before.